assets/js/phoenix_live_view/index.ts

```diff
@@ -146,6 +146,25 @@ export interface LiveSocketOptions {
       query: string,
       defaultQuery: () => Element[],
     ) => Element[];
+    /**
+     * When defined, called with a start callback that needs to be called
+     * to perform the actual patch. Failing to call the start callback causes
+     * the page to become stuck.
+     *
+     * This can be used to delay patches in order to perform view transitions,
+     * for example:
+     *
+     * ```javascript
+     * let liveSocket = new LiveSocket("/live", Socket, {
+     *   dom: {
+     *     beforePatch(start) {
```

Comment:
I am wondering if it should be called `beforePatch`, given it receives a function, which I assume does the whole patching, and therefore you can execute something afterwards? There is also onPatchStart and onPatchEnd, which could add to the confusion.

I'd suggest something like onDocumentPatchStart or onDocumentPatch, to make it clear this is invoked before patching the whole document. Or onTreePatchStart etc. Thoughts?